### ISSUE TITLE
Set 'Vendor' field in RPM packages

### ISSUE THREAD

**Issue by epa** [dotnet/core#3577]
Please could you tweak your RPM packages to include the `Vendor` field.  I believe you just need one line in the spec file:

`Vendor: https://github.com/dotnet/`

or you can call it whatever you wish.

**Comment by epa** [comment]
... or you could set `Packager` if you want.  Either is useful when checking what RPM packages are installed on a machine.

**Comment by scalablecory** [comment]
@dagood @leecow

**Comment by dagood** [comment]
What are you using to look at the vendor field, and what version of our packages? I'm not too familiar with these fields, but this command seems to show we do set both:

```
$ rpm -qa --qf "%{Name}: '%{Vendor}' '%{Packager}'\n" | grep dotnet-
dotnet-host: '.NET Foundation' '.NET Core Team <dotnetpackages@dotnetfoundation.org>'
dotnet-apphost-pack-3.0: '.NET Foundation' '.NET Core Team <dotnetpackages@dotnetfoundation.org>'
dotnet-runtime-deps-3.0: '.NET Foundation' '.NET Core Team <dotnetpackages@dotnetfoundation.org>'
dotnet-targeting-pack-3.0: '.NET Foundation' '.NET Core Team <dotnetpackages@dotnetfoundation.org>'
dotnet-hostfxr-3.0: '.NET Foundation' '.NET Core Team <dotnetpackages@dotnetfoundation.org>'
dotnet-runtime-3.0: '.NET Foundation' '.NET Core Team <dotnetpackages@dotnetfoundation.org>'
dotnet-sdk-3.0: '.NET Foundation' 'Microsoft <dotnetcore@microsoft.com>'
```

**Comment by epa** [comment]
Here's what I see when running that command:

```
$ rpm -qa --qf "%{Name}: '%{Vendor}' '%{Packager}'\n" | grep dotnet-
gpg-pubkey: '(none)' '@dotnet-sig_dotnet (None) <@dotnet-sig#dotnet@copr.fedorahosted.org>'
dotnet-templates-3.0: '(none)' '(none)'
dotnet-runtime-3.0: '(none)' '(none)'
dotnet-sdk-3.0: '(none)' '(none)'
dotnet-targeting-pack-3.0: '(none)' '(none)'
dotnet-hostfxr-3.0: '(none)' '(none)'
dotnet-host: '(none)' '(none)'
dotnet-apphost-pack-3.0: '(none)' '(none)'
```

**Comment by epa** [comment]
```
$ rpm -q fedora-release
fedora-release-29-11.noarch
$ rpm -qi dotnet-sdk-3.0
Name        : dotnet-sdk-3.0
Version     : 3.0.100
Release     : 4.fc29
Architecture: x86_64
Install Date: Thu 17 Oct 2019 12:01:42 PM BST
Group       : Unspecified
Size        : 142563728
License     : MIT and ASL 2.0 and BSD
Signature   : RSA/SHA1, Thu 17 Oct 2019 06:30:13 AM BST, Key ID 1e440b2b98e04be1
Source RPM  : dotnet3.0-3.0.100-4.fc29.src.rpm
Build Date  : Thu 17 Oct 2019 06:24:53 AM BST
Build Host  : copr-builder-966381471.novalocal
Relocations : (not relocatable)
URL         : https://github.com/dotnet/
Summary     : .NET Core 3.0 Software Development Kit
Description :
The .NET Core SDK is a collection of command line applications to
create, build, publish and run .NET Core applications.

.NET Core is a fast, lightweight and modular platform for creating
cross platform applications that work on Linux, Mac and Windows.

It particularly focuses on creating console applications, web
applications and micro-services.
```

**Comment by omajid** [comment]
>     Build Host  : copr-builder-966381471.novalocal

These are the .NET Core builds produced by the [Fedora .NET SIG](https://fedoraproject.org/wiki/SIGs/DotNet). That explains why @dagood and @epa are seeing different values of packagers. Please report these issues at https://pagure.io/dotnet-sig/packaging-issues/issues, or cc me (or another .NET SIG member) if you are filing issues in github

With my Fedora .NET SIG hat on, I don't think we should set a `Vendor` tag. For two reasons: [Fedora packaging guidelines forbid that](https://docs.fedoraproject.org/en-US/packaging-guidelines/#_tags_and_sections) (these packages are trying to shape up into what might go into Fedora proper) and there really isn't a vendor behind these packages other than the Fedora .NET Community.

For `Packager`, I guess we can put the Fedora .NET SIG, but I am still hesitant about adding tags that the packaging guidelines forbid.

**Comment by epa** [comment]
Thanks, I'll continue the discussion in the other bug tracker.  Or are there RPM packages built by this project I could switch to instead?  (Sorry to be thick... I don't see anything under https://github.com/orgs/dotnet/packages)

**Comment by epa** [comment]
@omajid I'm having trouble logging in to the pagure site; I understand from your reply it's OK to file bugs here as long as you're cc'd?

I understand that Fedora packages don't have a `Vendor` tag.  But that is exactly why I suggested adding one on these .NET Core packages, which aren't part of Fedora itself.  As a sysadmin I check the packages installed on a system and make sure they can be traced back to either the original distribution (Fedora) or a particular 3rd party repository.  If Vendor is set neither by the stock OS nor by 3rd party packages, it becomes a bit useless.  My suggestion would be to set the `Vendor` tag while the packages remain outside Fedora, and of course to unset it when they are included in a future Fedora release as part of the base system.

It doesn't really matter what the field is set to, I agree that the whole concept of a 'vendor' is a bit enterprisey and not a great fit for community projects, but you can just have `Fedora .NET Community` as the value.  Or anything at all, really, just something that can reliably be used to distinguish the packages from other repos and from the Fedora base system.

**Comment by omajid** [comment]
>  Or are there RPM packages built by this project I could switch to instead? 

Yeah, take a look at https://dotnet.microsoft.com/download/linux-package-manager/fedora29/sdk-current

Let me think about this `Vendor`/`Packager` some more.

**Comment by epa** [comment]
Those are the commands I ran, and it installed the packages built by the Fedora .NET SIG!

I think the problem is that I have that repository in `/etc/yum.repos.d` as well as the Microsoft one.  When both provide the same packages, `yum` has some kind of rule that ends up preferring the .NET SIG.

I removed that repository:

`sudo rm /etc/yum.repos.d/_copr\:copr.fedorainfracloud.org\:group_dotnet-sig\:dotnet.repo`

and now following the instructions on the page you linked gives the expected packages.

**Comment by dagood** [comment]
Thanks @omajid for clearing this up!

Looking a bit at ways of making this clearer:

My first thought was adding steps to the Msft instructions to explicitly turn off the .NET SIG repo to make sure people end up with the "right" package when they run the instructions. My first impression is that seems pretty unfriendly, and extending the idea to remove all non-Msft distros of .NET Core doesn't seem reasonable. (Don't know if the full list is even known, and it will only get bigger as it becomes more reasonable to build .NET Core from source. (Which will be awesome. 🙂))

Maybe we could include an optional `rpm --qf "%{VENDOR}, %{PACKAGER}\n" -q dotnet-sdk-3.0` step to check if the package is Microsoft's?

But both of those would just be clutter for nearly everyone running the steps. 😕 

Even though this confusion somewhat derails initial investigations, it is pretty rare, so I'm not sure it's worth doing anything big. Also, to some degree, we do need to rely on people knowing how they have their machines set up. Something making it harder to keep track of might be trying out .NET via COPR in Fedora a long time ago, then trying again via the Msft repo now.

I added a note to https://github.com/dotnet/core/pull/2848 that when I can get around to finishing it, I should include this check as a standard diagnostic step for users reporting issues with packages.

Closing because this particular issue is resolved. Curious if anyone has thoughts on the above.
